### PR TITLE
Integration tests for import/include

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

We need more tests for `include/import`. This is them.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
`include_role`
`include_task`

`import_play`
`import_task`
`import_role`

##### ANSIBLE VERSION

```
2.5
```

### CODE REVIEW

File: test/integration/targets/include_import/play/test_import_play.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+- hosts: localhost
+  connection: local
```
Comment (bcoca) [inline test/integration/targets/include_import/play/test_import_play.yml]: connection: local should be redundant

File: test/integration/targets/include_import/role/test_import_role.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    - name: Pass variable to role
+      include_role:
+        name: role1
+        tasks_from: vartest.yml
+      vars:
+        where_am_i_defined: in the task
+
+    # FIXME Currently failing
+    # Not sure if this is suppsode to work
```
Comment (bcoca) [inline test/integration/targets/include_import/role/test_import_role.yml]: it should work

File: test/integration/targets/include_import/role/test_import_role.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    - name: Test including vars file and tasks file from a role
+      include_role:
+        name: role3
+        tasks_from: vartest.yml
+        vars_from: role3vars.yml
+        private: no
+
+    - name: Use variable for role name
+      include_role:
+        name: "{{ role_name }}"
```
Comment (bcoca) [inline test/integration/targets/include_import/role/test_import_role.yml]: this checks play level var, we also need to check host level ones

File: test/integration/targets/include_import/play/test_import_play.yml (hunk 1 above)
Comment (samdoran) [reply test/integration/targets/include_import/play/test_import_play.yml]: Fixed

File: test/integration/targets/include_import/role/test_import_role.yml (hunk 2 above)
Comment (samdoran) [reply test/integration/targets/include_import/role/test_import_role.yml]: Currently failing. I'll leave it in but commented out for now.

**Comment by abadger** [comment]
@sdoran could we merge this now and add more tests in a separate PR later?

**Comment by abadger** [comment]
Sorry sdoran......  @samdoran, question was for you: Could we merge this now and add/enable more tests in a later PR?

**Comment by samdoran** [comment]
@abadger Yes, that would probably be wise. It already contains a good number of tests though I know we need more.

Let me use @bcoca's suggestion of tagging things in order to skip the failing tests for now rather than commenting them out. I'll push a commit shortly.

**Comment by samdoran** [comment]
These aren't exhaustive and I'm sure I've missed some things, but I believe these are good enough to merge for now.

**Comment by abadger** [comment]
rebuild_merge

**Comment by mattclay** [comment]
CI failure due to consistently "unstable" test on all platforms:

https://app.shippable.com/github/ansible/ansible/runs/48928/48/tests

**Comment by samdoran** [comment]
Switched to using `fail` module. Hopefully that will correct the "unstable" tests.

File: test/integration/targets/include_import/roles/role1/tasks/fail.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+- fail:
+    msg: This command should always fail
```
Comment (mattclay) [inline test/integration/targets/include_import/roles/role1/tasks/fail.yml]: The task name must contain `EXPECTED FAILURE` for this to pass CI.

Comment (samdoran) [reply test/integration/targets/include_import/roles/role1/tasks/fail.yml]: Thanks Matt! I'll get that fixed.

**Comment by jctanner** [comment]
@samdoran can you rebase? Was there also another test that was still broken? Can you point it out?

**Comment by samdoran** [comment]
Latest failure was transient and unrelated. Just re-ran and it passed. I’ll rebase in a bit.

**Comment by abadger** [comment]
rebuild_merge

**Comment by samdoran** [comment]
@jctanner There are a few tests failing that are either commented out or just skipped.

[Variables from a role with `private: no` do not get passed to the play](https://github.com/ansible/ansible/pull/33418/files#diff-8d237e199e8dd6fa6e4ac39886bab593R74)
[Passing an entire dictionary of variables to a role](https://github.com/ansible/ansible/pull/33418/files#diff-8d237e199e8dd6fa6e4ac39886bab593R74)

**Comment by abadger** [comment]
Merged to devel and cherrypicked for the 2.4.3 rc3 release.

**Pull request merged**